tomopy/misc/npmath.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy  as np
+import pandas as pd
+import logging
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+__author__ = "Chen Zhang"
+__all__    = [ 'gauss',
+               'calc_cummulative_dist',
+               'calc_affine_transform',
+]
+
+
+def gauss(x, *p):
+    """
+    simple Guassian func used for quick curve fit
+
+    Parameters
+    ----------
+    x  :  np.1ndarray
+        1D array for curve fitting
+    p  :  parameter lis t
+        A,          mu,      sigma  =   p
+        magnitude  center     std
+
+    Returns
+    -------
+    1d Gaussian distrobution evaluted at x with p
+    """
+    A, mu, sigma = p
+    return A*np.exp(-(x-mu)**2/(2.*sigma**2))
+
+
+def calc_cummulative_dist(data, label=None, steps=None):
+    """
+    Calculated the cumulative distribution from a data array without using
+    binning.  
+    This provides a more robust way to representing the statistics of the 
+    incoming data without worrying about the binning/samping effect.
+
+    Parameters
+    ----------
+    data  :  1D np.array | full pd.DataFrame
+        1-D numpy array or dataFrame (when label is provided)
+    label :  [ None | str ], optional
+        column label for analyzed data
+    steps :  [ None | int ], optional
+        Number of elements in the returning array
+
+    Returns
+    -------
+    pltX  : 1D np.array
+        plt data along x (data direction)
+    pltY  : 1D np.array
+        plt data long y (density direction)
+    """
+    if isinstance(data, pd.DataFrame):
```

Comment:
It looks like you only use `pandas` here on this one line. We try to minimize the number of dependencies that `tomopy` needs because it make installations smaller and may improve compatibility. Can you explain why your new feature requires the installation of `pandas`?

Reply:
Based on my experience, Pandas can be very handy when it comes to statistic analysis with a large dataset with many features/labels, which is something difficult to do with barebone numpy.  I would even recommend Tomopy to use pandas to handle tabulated data internally if possible.  However, if this is something that goes against Tomopy development policy, I can remove this function from the module.

Reply:
@KedoKudo : You are describing a feature enhancement for tomopy.  Sounds worthwhile.  Is it necessary for this PR?  Can the `pandas` requirement be separated so the merits can be discussed independent of this PR?

Reply:
It's not against development policy to introduce new packages, but in this particular case, it looks like the only reason you want to add `pandas` as a dependency is so you can check whether or you need to index data. I think you should refactor this way:
```python
# yours which requires pandas
calc_cummulative_dist(data, label='foo')

# refactored without pandas dependency
calc_cummulative_dist(data['foo'])
```

Reply:
@prjemian : That's a good point.  I decoupled the function dependencies on pandas.